lang/cs/Org.Apache.REEF.Tests/Functional/ReefFunctionalTest.cs

```diff
@@ -263,10 +223,42 @@ protected void PeriodicUploadLog(object source, ElapsedEventArgs e)
             }
         }
 
-        protected string GetLogFile(string logFileName, string subfolder = "driver", string testFolder = DefaultRuntimeFolder)
+        internal string[] ReadLogFile(string logFileName, string subfolder = "driver", string testFolder = DefaultRuntimeFolder)
+        {
+            string fileName = string.Empty;
+            string[] lines = null;
+            for (int i = 0; i < 60; i++)
+            {
+                try
+                {
+                    fileName = GetLogFileName(logFileName, subfolder, testFolder);
+                    lines = File.ReadAllLines(fileName);
+                    break;
+                }
+                catch (Exception e)
+                {
+                    if (i == 59)
+                    {
+                        // log only last exception before failure
+                        Logger.Log(Level.Verbose, e.ToString());
+                    }
+                    if (i < 59)
+                    {
+                        Thread.Sleep(SleepTime);
+                    }
+                }
+            }
+            if (lines == null)
+            {
+                Assert.True(false, "Cannot read from log file " + fileName);
```

Comment:
We used to use `Assert.Fail` when we had mstest, but xUnit doesn't have it. The best I can think of is `Assert.True(lines != null, ...)` (a more straightforward `Assert.NotNull` doesn't allow to add a message)